example_with_targets/src/lib.rs

```diff
@@ -0,0 +1,27 @@
+use shopify_function::prelude::*;
+use shopify_function::Result;
+
+#[shopify_function_target(
+  query_path = "a.graphql",
+  schema_path = "schema.graphql",
+  output_result_type = FunctionAResult
+)]
+fn a(_input: input::ResponseData) -> Result<output::FunctionAResult> {
+    Ok(output::FunctionAResult {
+        status: Some(200),
+    })
+}
+
+#[shopify_function_target(
+  query_path = "b.graphql",
+  schema_path = "schema.graphql",
+  output_result_type = FunctionBResult
+)]
+fn b(input: input::ResponseData) -> Result<output::FunctionBResult> {
+    Ok(output::FunctionBResult {
+        name: Some(format!("new name: \"{}\"", input.id)),
+    })
+}
```

Comment:
> I wonder if there's a bit too much "magic" here.
>
> When reading this initially, it makes it seem as though the input to the two functions is the same, but that is not the case IIUC.

😆 Fair! I also thought to leave the original functions outside the module as well since it's a bit weird to have it disappear.

> I wonder if we'd be better off having the application define the modules explicitly

I'm not sure I'd want to tell people to set up separate modules via docs 🤔